modules/socat/options.go

```diff
@@ -0,0 +1,76 @@
+package socat
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+type options struct {
+	// Targets is the map of targets of the socat container
+	Targets    map[int]Target
+	targetsCmd string
+}
+
+func defaultOptions() options {
+	return options{
+		Targets: map[int]Target{},
+	}
+}
+
+// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options)
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// Target represents a target for the socat container.
+// Create a new target with NewTarget or NewTargetWithInternalPort.
+type Target struct {
+	exposedPort  int
+	internalPort int
+	host         string
+}
+
+func NewTarget(exposedPort int, host string) Target {
+	return Target{
+		exposedPort: exposedPort,
+		host:        host,
+	}
+}
+
+func NewTargetWithInternalPort(exposedPort int, internalPort int, host string) Target {
+	return Target{
+		exposedPort:  exposedPort,
+		internalPort: internalPort,
+		host:         host,
+	}
+}
+
+// WithTargets sets the targets for the socat container.
+// The host of each target must be without the port, as it is internally mapped to the exposed port.
+func WithTargets(targets ...Target) Option {
```

Comment:
question: would it be better to use multiple WithTarget aor WithTargetInternalPort?

Reply:
this is one of the dilemmas I recently have: allow variadic arguments to do the job, or pass the same option multiple times... 

I do not have a strong opinion. The variadic argument is more concise Vs the explicitness of the multi-usage of the option.

What are your thoughts on this?

Reply:
On the other hand, I'm looking at #3070, and for those structured types, I see no issue in passing them as variadic arguments instead 🤔 

I'm fine with keeping the single argument here for the targets, as it's very scoped to the socat container. But for the other PR, I see the value in adding them as they are passed as collections